scripts/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,243 @@
+    'tilt lab swear uncle prize favorite river myth assault transfer venue soap lady someone marine reject fork brain swallow notice glad salt sudden pottery',
+  ],
+  pubkeys: [
+    'xpub6CvvCBtHqFfErbcW2Rv28TmZ3MqcFuWQVKGg8xDzLeAwEAHRz9LBTgSFSj7B99scSvZGbq6TxAyyATA9b6cnwsgduNs9NGKQJnEQr3PYtwK',
+    'xpub6CA16g2qPwukWAWBMdJKU3p2fQEEi831W3WAs2nesuCzPhbrG29aJsoRDSEDT4Ac3smqSk51uuv6oujU3MAAL3d1Nm87q9GDwE3HRGQLjdP',
+    'xpub6BwNT613Vzy7ARVHDEpoX23SMBEZQMJXdqTWYjQKvJZJVDBjEemU38exJEhc6qbFVc4MmarN68gUKHkyZ3NEgXXCbWtoXXGouHpwMEcXJLf',
+    'xpub6DCyPHg4AwXsdiMh7QSTHR7afmNVwZKHBBMFUiy5aCYQNaWp68ceQXYXCGQr5fZyLAe5hiJDdXrq6w3AXzvVmjFX9F7EdM87repxJEhsmjL',
+    'xpub6CgPUcCCJ9pAK7Rj52hwkxTutSRv91Fq74Hx1SjN62eg6Mp3S3YCJFPChPaDjpp9jCbCZHibBgdKnfNdq6hE9umyjyZKUCySBNF7wkoG4uK',
+  ],
+  walletConfig: {
+    pubkeys: [],
+    total: 5,
+    minSignatures: 3,
+  }
+};
+multisigWalletsData.walletConfig.pubkeys = multisigWalletsData.pubkeys;
+
+export class WalletPrecalculationHelper {
+  WALLETS_FILENAME = '';
+
+  walletsDb = [];
+
+  /**
+   * Initializes the helper with a filename to sync the local wallet storage with.
+   * @param walletsFilename
+   */
+  constructor(walletsFilename) {
+    this.WALLETS_FILENAME = walletsFilename || './tmp/wallets.txt';
+  }
+
+  /**
+   * Generates 22 addresses for a wallet 24-word seed.
+   * @param [params]
+   * @param {string} [params.words] Optional wallet seed words. If empty, generates a new wallet
+   * @param {number} [params.addressIntervalStart=0] Optional interval start index
+   * @param {number} [params.addressIntervalEnd=22] Optional interval end index
+   * @param {{minSignatures:number,wordsArray:string[]}} [params.multisig] Optional multisig object
+   * @returns {{addresses: string[], words: string}}
+   */
+  static generateAddressesForWordsSeed(params = {}) {
+    const timeStart = Date.now().valueOf();
+
+    // Generating seed if none was informed
+    let wordsInput = params.words;
+    if (!wordsInput) {
+      wordsInput = wallet.generateWalletWords();
+    }
+
+    // Calculating addresses
+    const addressIntervalStart = params.addressIntervalStart || 0;
+    const addressIntervalEnd = params.addressIntervalEnd || 22;
+    const addressesArray = [];
+    let multisigDebugData = null;
+    if (params.multisig) {
+      // Multisig calculation
+      const pubkeys = params.multisig.wordsArray.map(w => walletUtils.getMultiSigXPubFromWords(w));
+      for (let i = addressIntervalStart; i < addressIntervalEnd; ++i) {
+        const redeemScript = walletUtils.createP2SHRedeemScript(
+          pubkeys,
+          params.multisig.minSignatures,
+          i
+        );
+        const address = Address.payingTo(Script.fromBuffer(redeemScript), config.network);
+        addressesArray.push(address.toString());
+
+        // Informing debug data
+        multisigDebugData = {
+          words: wordsInput,
```

Comment:
Why that? For multisig you use the array of words

Reply:
Removed it on [3c6e4fc](https://github.com/HathorNetwork/hathor-wallet-headless/pull/189/commits/3c6e4fc97e977eaf1e5a571d25742a19b67414c0)